### ISSUE TITLE
import block creates an unnecessary dependency to default provider

### ISSUE THREAD

**Issue by lgonzalez-silen** [hashicorp/terraform#34328]
### Terraform Version

```shell
Terraform v1.6.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google-beta v5.7.0
```

### Terraform Configuration Files

```terraform
provider "google-beta" {
  project = "myproject"
}

data "google_project" "project" {
  provider = google-beta
}

import {
  provider = google-beta
  id       = "mydbinstance"
  to       = google_sql_database_instance.db_instance
}

resource "google_sql_database_instance" "db_instance" {
  provider         = google-beta
  name             = "mydbinstance"
  database_version = "POSTGRES_14"

  settings {
    tier = "db-custom-2-8192"
  }

  lifecycle {
    ignore_changes = [timeouts, settings]
  }
}
```

### Debug Output

https://gist.github.com/lgonzalez-silen/c4d1fcbedd5e274e04eca9d195edd364

### Expected Behavior

Since I specified a beta provider to use with the import block, I don't expect the default provider to be a dependency and be pulled in on `terraform init`, or `terraform plan`, if adding the import block later on.

If I add the import block and resource `google_sql_database_instance.db_instance` after init, I expect it to say

```
Apply complete! Resources: 1 imported, 0 added, 0 changed, 0 destroyed.
```

on first run, and then

```
No changes. Your infrastructure matches the configuration.
```

on subsequent runs, without asking me to upgrade and add the parent/default provider.

### Actual Behavior

```
│ Error: Inconsistent dependency lock file
|
│ The following dependency selections recorded in the lock file are inconsistent with the current configuration:
│   - provider registry.terraform.io/hashicorp/google: required by this configuration but no version is selected
│ 
│ To update the locked dependency selections to match a changed configuration, run:
│   terraform init -upgrade
```

### Steps to Reproduce

1. Comment out the import block and resource `google_sql_database_instance.db_instance` in the config above.
2. `terraform init`
3.  Note that only google-beta is installed
4. `terraform apply`
5. Uncomment the import block and resource `google_sql_database_instance.db_instance`
6. `terraform apply`
7. Note that it now needs to install the google provider
8. `terraform init -upgrade`
9. `terraform apply`
10. Hack file `infra/poc/.terraform.lock.hcl` to remove the google provider; leave the google-beta one alone.
11. Delete `.terraform/providers/registry.terraform.io/hashicorp/google`
12. `terraform apply`
13. Note the error
14. Comment out the import block
15. `terraform apply`
16. Now there's no error, demonstrating the google provider was never needed in the first place

**Comment by kmoe** [comment]
@lgonzalez-silen This is fixed in Terraform v1.7 - I couldn't reproduce it with the latest 1.7.0 dev build, but could with v1.6.5. Would you mind trying with the latest [v1.7.0-alpha20231130](https://releases.hashicorp.com/terraform/1.7.0-alpha20231130/) version to confirm?

**Comment by lgonzalez-silen** [comment]
@kmoe confirmed fixed in [1.7.0-beta1](https://releases.hashicorp.com/terraform/1.7.0-beta1/)

thank you!
